ef_tx.go

```diff
@@ -0,0 +1,68 @@
+package bux
+
+import (
+	"context"
+	"encoding/hex"
+
+	"github.com/libsv/go-bt/v2"
+)
+
+func ToEfHex(ctx context.Context, tx *Transaction, store TransactionGetter) (efHex string, ok bool) {
+	btTx := tx.parsedTx
+
+	if btTx == nil {
+		var err error
+		btTx, err = bt.NewTxFromString(tx.Hex)
+		if err != nil {
+			return "", false
+		}
+	}
+
+	needToHydrate := false
+	for _, input := range btTx.Inputs {
+		if input.PreviousTxScript == nil {
+			needToHydrate = true
+			break
+		}
+	}
+
+	if needToHydrate {
+		if ok := hydrate(ctx, btTx, store); !ok {
+			return "", false
+		}
+	}
+
+	return hex.EncodeToString(btTx.ExtendedBytes()), true
+}
+
+func hydrate(ctx context.Context, tx *bt.Tx, store TransactionGetter) (ok bool) {
+	txToGet := make([]string, 0, len(tx.Inputs))
+
+	for _, input := range tx.Inputs {
+		txToGet = append(txToGet, input.PreviousTxIDStr())
+	}
+
+	parentTxs, err := store.GetTransactionsByIDs(ctx, txToGet)
+	if err != nil {
+		return false
+	}
+	if len(parentTxs) != len(tx.Inputs) {
+		return false
+	}
+
+	for _, input := range tx.Inputs {
+		prevTxId := input.PreviousTxIDStr()
```

Comment:
Minor - lint error:
`var-naming: var prevTxId should be prevTxID (revive)go-golangci-lint`

Reply:
Done